crates/forge_loader/src/manifest.rs

```diff
@@ -25,24 +26,68 @@ pub struct FunctionMod<'a> {
     providers: Option<AuthProviders<'a>>,
 }
 
+// Modified 
 #[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
 struct ModInfo<'a> {
+    function: &'a str,
+}
+
```

Comment:
If you notice, a lot of modules basically contain at least these three properties: key, function, resolver.function, where the function and resolver properties are optional. You could factor this into a common struct like so:
```rust
#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
struct CommonKeys<'a> {
    key: &'a str,
    #[serde(borrow)]
    function: Option<&'a str>,
    #[serde(borrow)]
    resolver: Option<ModInfo<'a>>,
}
```
and then embed it into the other structs like so:

```rust
#[derive(Default, Debug, Clone, PartialEq, Eq, Deserialize)]
struct MacroMod<'a> {
    #[serde(flatten, borrow)]
    common_keys: CommonKeys<'a>,
    ... other keys here
}
```

Reply:
This makes sense to me to simplify the implementation, but how would rust know the mapping? To me, setting common_keys looks like setting a struct within a struct. And keys, function, and resolver would be the first level of the struct.

Reply:
The `serde(flatten)` attribute will basically just treat the struct's keys as being embedded in the parent struct. See https://serde.rs/attr-flatten.html for more details.